### PR TITLE
DOC: Update release notes for v0.7.0.

### PR DESCRIPTION
**Pull request by danielballan**

### CODE REVIEW

File: doc/source/release_notes.rst [hunk 1]
```diff
@@ -55,6 +41,109 @@ API Changes
...
+
+  .. code-block:: none
+  
+      In [5]: motor = EpicsMotor('XF:31IDA-OP{Tbl-Ax:X1}Mtr', name='motor')
+  
+      In [6]: print(motor)
+      data keys (* hints)
+      -------------------
+      *motor
+      motor_user_setpoint
```
Comment (tacaswell) [inline doc/source/release_notes.rst]: There is a missing space here.

Comment (danielballan) [inline doc/source/release_notes.rst]: fixed

File: doc/source/release_notes.rst [hunk 2]
```diff
@@ -55,6 +41,121 @@ API Changes
       :func:`ophyd.setup_ophyd`, which is still available as a top-level
       import.
 
+* For complex devices, the same child device may be used in both ``read_attrs``
```
Comment (tacaswell) [inline doc/source/release_notes.rst]: This is missing "When recursing through complex devices ``read`` in no longer called as part of ``read_configuration``"

File: doc/source/release_notes.rst [hunk 3]
```diff
@@ -55,6 +41,121 @@ API Changes
...
+Enhancements
+************
+
+* Many devices picked up a new ``hints`` property. Its goal is to highlight the
+  most interesting or important fields---often a small subset of all the fields
+  that are read---in support of automated visualization and processing. It does
+  not affect what is read or recorded; nothing is permanently altered or lost
+  if the hints are incorrect. The content of hints may be changed in future
+  releases, as this feature is experimental. Movable devices return a
+  dictionary with the key ``dimensions`` mapped to a list of field names,
```
Comment (tacaswell) [inline doc/source/release_notes.rst]: I thought 'dimensions' was a plan level thing?  I don't think anything in ophyd actually returns a anything but `{'fields': [...]}`

Comment (danielballan) [inline doc/source/release_notes.rst]: You're right; I wasn't thinking straight.

**Pull request merged**